pilot/cmd/pilot-agent/security/security.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	securityModel "istio.io/istio/pilot/pkg/security/model"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/jwt"
+	"istio.io/istio/pkg/security"
+	"istio.io/istio/security/pkg/credentialfetcher"
+	"istio.io/istio/security/pkg/nodeagent/plugin/providers/google/stsclient"
+	"istio.io/pkg/log"
+)
+
+func SetupSecurityOptions(proxyConfig meshconfig.ProxyConfig, jwtPolicy, clusterIDVar, podNamespaceVar,
```

Comment:
I don't think having 16 parameters to a function is a good idea

Reply:
Also, I am not really sure what the point is. We are basically passing 16 parameters into this function then the function just puts them in a struct. Why not put them in the struct directly?

Reply:
The code has been revised to put the parameters in the struct directly.